src/Util/InstallationUtil.php

```diff
@@ -58,124 +58,113 @@
-            } else {
-                throw new BunqException(self::ERROR_CANNOT_CREATE_API_KEY_PRODUCTION);
-            }
+        if (is_null($environmentType)) {
+            $environmentType = BunqEnumApiEnvironmentType::SANDBOX();
+        } else {
+            // Environment already passed
+        }
 
-            $methodInitializeInstallationContext = static::createAccessibleReflectionMethod(
-                ApiContext::class,
-                self::METHOD_INITIALIZE_INSTALLATION_CONTEXT
-            );
-            $methodInitializeInstallationContext->invoke($context);
+        static::setPrivateProperty($context, self::PROPERTY_ENVIRONMENT_TYPE, $environmentType);
 
-            $methodRegisterDevice = static::createAccessibleReflectionMethod(
+        if ($environmentType->equals(BunqEnumApiEnvironmentType::SANDBOX()) && is_null($apiKey)) {
+            $methodCreateSandboxUser = static::createAccessibleReflectionMethod(
                 ApiContext::class,
-                self::METHOD_REGISTER_DEVICE
+                self::METHOD_CREATE_SANDBOX_USER
             );
-            $methodRegisterDevice->invoke($context, gethostname(), []);
 
-            $methodInitializeSessionContext = static::createAccessibleReflectionMethod(
-                ApiContext::class,
-                self::METHOD_INITIALIZE_SESSION_CONTEXT
-            );
-            $methodInitializeSessionContext->invoke($context);
+            $methodCreateSandboxUser->invoke($context);
+        } elseif (!is_null($apiKey)) {
+            static::setPrivateProperty($context, self::PROPERTY_API_KEY, $apiKey);
+        } else {
+            throw new BunqException(self::ERROR_CANNOT_CREATE_API_KEY_PRODUCTION);
+        }
 
-            if ($contextFileName === null) {
-                $context->save();
-            } else {
-                $context->save($contextFileName);
-            }
-        } catch (BunqException $exception) {
-            echo sprintf(self::ERROR_BUNQ_EXCEPTION, $exception->getMessage());
-            var_dump($exception);
-        } catch (Exception $exception) {
-            echo sprintf(self::ERROR_EXCEPTION, $exception->getMessage());
+        $methodInitializeInstallationContext = static::createAccessibleReflectionMethod(
+            ApiContext::class,
+            self::METHOD_INITIALIZE_INSTALLATION_CONTEXT
+        );
+        $methodInitializeInstallationContext->invoke($context);
+
+        $methodRegisterDevice = static::createAccessibleReflectionMethod(
+            ApiContext::class,
+            self::METHOD_REGISTER_DEVICE
+        );
+        $methodRegisterDevice->invoke($context, gethostname(), []);
+
+        $methodInitializeSessionContext = static::createAccessibleReflectionMethod(
+            ApiContext::class,
+            self::METHOD_INITIALIZE_SESSION_CONTEXT
+        );
+        $methodInitializeSessionContext->invoke($context);
+
+        if ($contextFileName === null) {
```

Comment:
According to the method signature, this could never be null 😅 

TODO: remove if statement after #137